compiler/codegen/OMRCodeGenPhase.cpp

```diff
@@ -419,39 +419,43 @@ void
 OMR::CodeGenPhase::performSetupForInstructionSelectionPhase(TR::CodeGenerator * cg, TR::CodeGenPhase * phase)
    {
    TR::Compilation *comp = cg->comp();
-   // TODO (GuardedStorage)
-#if defined(OMR_GC_CONCURRENT_SCAVENGER)
-   traceMsg(comp, "GuardedStorage: in performSetupForInstructionSelectionPhase\n");
 
-   auto mapAllocator = getTypedAllocator<std::pair<TR::TreeTop*, TR::TreeTop*> >(comp->allocator());
-
-   std::map<TR::TreeTop*, TR::TreeTop*, std::less<TR::TreeTop*>, TR::typed_allocator<std::pair<TR::TreeTop*, TR::TreeTop*>, TR::Allocator> >
-      currentTreeTopToappendTreeTop(std::less<TR::TreeTop*> (), mapAllocator);
-
-   for (TR::PreorderNodeIterator iter(comp->getStartTree(), comp); iter != NULL; ++iter)
+   if (cg->isConcurrentScavengeEnabled())
```

Comment:
@0xdaryl as stated in the commit message this is a pre-emptive change. The GC currently does not have a runtime flag which determines whether concurrent scavenge is enabled or not, i.e. the query which determines the concurrent scavenge GC policy.

As such the `isConcurrentScavengeEnabled` is currently guarded by a compile time flag. When GC implements a runtime query for the concurrent scavenge GC policy `isConcurrentScavengeEnabled` will change accordingly. The "ENABLED" part you mention above will be implemented as such.

The "SUPPEROTED" part is implmented already by `getS390SupportsGuardedStorageFacility` query which checks for processor support. The value returned by this call is a cached value obtained from the port library by extracting STFLE bits from the processor. Once the GC implements their runtime check they too will query the port library in a similar fashion so the GC and JIT are both synchronized with the value each respective library returns for a "isConcurrentScavengeEnabled" query.

Hope this clears things up a bit. Did I answer your question?